include/boost/url/query_param.hpp

```diff
@@ -298,12 +298,12 @@ struct query_param
     {
         if(has_value)
             return {
-                pct_encoded_view(key),
-                pct_encoded_view(value),
+                decode_view(key),
+                decode_view(value),
```

Comment:
same problem here where we are validating needlessly, and possibly throwing from a function marked `noexcept`